pilot/pkg/serviceregistry/kube/controller/endpoints.go

```diff
@@ -194,8 +194,8 @@ func (e *endpointsController) buildIstioEndpoints(endpoint interface{}, host hos
 	ep := endpoint.(*v1.Endpoints)
 	for _, ss := range ep.Subsets {
 		for _, ea := range ss.Addresses {
-			pod := getPod(e.c, ea.IP, &metav1.ObjectMeta{Name: ep.Name, Namespace: ep.Namespace}, ea.TargetRef, host)
-			if pod == nil {
+			pod, expectedPod := getPod(e.c, ea.IP, &metav1.ObjectMeta{Name: ep.Name, Namespace: ep.Namespace}, ea.TargetRef, host)
```

Comment:
The core of the issue is if there is no pod, we always continue. We used to only skip if PodRef is present and pod == nil, which is the restored behavior here. We want to skip only if there is no pod *and* we know there *must* be an associated pod - we just havent seen it yet